### PR TITLE
Use fake data in tests, add relationships in entities

### PR DESCRIPTION
**Pull request by JustD4nTe**
- Add missing table in db => TeamInLeague
- Use fake data in integrations and unit tests 
- Remove unused matches' endpoint (history of league)

### CODE REVIEW

File: src/SoccerStatistics.Api.Core/DTO/StadiumDTO.cs [hunk 1]
```diff
@@ -15,6 +17,6 @@ public class StadiumDTO
         public uint Lighting { get; set; }
         public string Architect { get; set; }
         public bool IsNational { get; set; }
-        public TeamBasicDTO Team { get; set; }
+        public IEnumerable<TeamBasicDTO> Teams { get; set; }
```
Comment (PieterQQ) [inline src/SoccerStatistics.Api.Core/DTO/StadiumDTO.cs:20]: I think stadium has only 1 representative team. Do we need Ienumerable here?

File: src/SoccerStatistics.Api.Database/Entities/Stadium.cs [hunk 2]
```diff
@@ -32,5 +33,6 @@ public class Stadium
         public string Architect { get; set; }
         [Required]
         public bool IsNational { get; set; }
+        public IEnumerable<Team> Teams { get; set; }
```
Comment (PieterQQ) [inline src/SoccerStatistics.Api.Database/Entities/Stadium.cs:36]: Same here

File: src/SoccerStatistics.Api.Database/Repositories/Interfaces/IMatchRepository.cs [hunk 3]
```diff
@@ -1,12 +1,10 @@
 ﻿using SoccerStatistics.Api.Database.Entities;
-using System.Collections.Generic;
 using System.Threading.Tasks;
 
 namespace SoccerStatistics.Api.Database.Repositories.Interfaces
 {
     public interface IMatchRepository : IRepository
     {
         Task<Match> GetByIdAsync(uint id);
-        Task<IEnumerable<Match>> GetHistoryOfMatchesByLeagueId(uint leagueId);
```
Comment (PieterQQ) [inline src/SoccerStatistics.Api.Database/Repositories/Interfaces/IMatchRepository.cs:10]: We dont want/need History of matches as a separate task?

**Review by PieterQQ — APPROVED** [review]

File: src/SoccerStatistics.Api.Core/DTO/StadiumDTO.cs (hunk 1 above)
Comment (JustD4nTe) [reply src/SoccerStatistics.Api.Core/DTO/StadiumDTO.cs:20]: This collection includes teams that play "at home" and do not own the stadium. If necessary, we can add OwnerTeam property.

File: src/SoccerStatistics.Api.Database/Repositories/Interfaces/IMatchRepository.cs (hunk 3 above)
Comment (JustD4nTe) [reply src/SoccerStatistics.Api.Database/Repositories/Interfaces/IMatchRepository.cs:10]: Yes! But we have to talk more about it in the meeting

Comment (Blazevarjo) [reply src/SoccerStatistics.Api.Database/Repositories/Interfaces/IMatchRepository.cs:10]: Filtering and sorting ftw

File: src/SoccerStatistics.Api.Application/Handlers/GetAllTeamsHandler.cs [hunk 4]
```diff
@@ -17,7 +17,7 @@ public GetAllTeamsHandler(ITeamService service)
             _service = service;
         }
 
-        public async Task<IEnumerable<TeamBasicDTO>> Handle(GetAllTeamsQuery request, CancellationToken cancellationToken) 
+        public async Task<IEnumerable<TeamDTO>> Handle(GetAllTeamsQuery request, CancellationToken cancellationToken) 
             => await _service.GetAllAsync();
     }
```
Comment (Blazevarjo) [inline src/SoccerStatistics.Api.Application/Handlers/GetAllTeamsHandler.cs:22]: Why switch it to more complex dto?

**Review by Blazevarjo — APPROVED** [review]

File: src/SoccerStatistics.Api.Application/Handlers/GetAllTeamsHandler.cs (hunk 4 above)
Comment (JustD4nTe) [reply src/SoccerStatistics.Api.Application/Handlers/GetAllTeamsHandler.cs:22]: While getting all leagues and stadiums we have complex dto too. That's the only reason why I changed it.

Comment (JustD4nTe) [reply src/SoccerStatistics.Api.Application/Handlers/GetAllTeamsHandler.cs:22]: I can revert changes to previous state :p But we should talk about this at the next meeting with our UI/UX designer (what specific information we store in list of teams in our apps)

**Pull request merged**